tox.ini

```diff
@@ -25,17 +24,25 @@ extras = all
 passenv = KRB5CCNAME, DISPLAY, https_proxy, USER, GIT_SSH_COMMAND, HOME, HOC_LIBRARY_PATH
 deps =
     neuron
-    pytest-xdist
+    pytest>=7.3.1
+    pytest-cov>=4.1.0
+    pytest-timeout>=2.1.0
+    pytest-xdist>=3.3.1  # multiprocessing
+    pytest-forked>=1.6.0  # isolation
+download = true
+allowlist_externals =
+    make
+    coverage
+coverage_options = --cov-report=xml --cov-config=.coveragerc --cov={[base]name} --cov=tests
+setenv =
+    NEURON_MODULE_OPTIONS='-nogui'
+    OMP_NUM_THREADS=1
 commands =
+    make clean
     nrnivmodl tests/test_data/mechanisms
-    python -m unittest discover ./tests
-    pytest -s -n 8 tests
-
-
-[testenv:check-version]
-skip_install = true
-deps = bbp-nse-ci
-commands = do_release.py -p . check-version
+    pytest {[testenv]coverage_options} tests --numprocesses=auto --timeout=600 -vsx --forked
```

Comment:
Do we have a reason for doing the tests in forked processes? I think BlueCelluLab needed this to have a 'clean' neuron for each tests, and avoid one test corrupting neuron states of another test.

Reply:
If it makes tests faster, it's cool though

Reply:
can we keep it just to be sure? I think @anilbey mentioned to me that BluePyOpt also need this and I think we use it in our test, what do you think?

Reply:
or maybe not, I don't see it in the tox of BluePyOpt

Reply:
and also our previous tests pass without any issue

Reply:
ok I'll remove it 👍